modules/programs/nushell.nix

```
@@ -21,40 +21,55 @@ in {
       description = "The package to use for nushell.";
     };
 
-    settings = mkOption {
-      type = with types;
-        let
-          prim = oneOf [ bool int str ];
-          primOrPrimAttrs = either prim (attrsOf prim);
-          entry = either prim (listOf primOrPrimAttrs);
-          entryOrAttrsOf = t: either entry (attrsOf t);
-          entries = entryOrAttrsOf (entryOrAttrsOf entry);
-        in attrsOf entries // { description = "Nushell configuration"; };
-      default = { };
+    config = mkOption {
+      type = types.nullOr types.path;
```

I would suggest having

```suggestion
      type = with types; nullOr (either str path);
```

and allow either a path to a file or a Nix string.

---

With Nix string you mean the contents of the file or a string of a path?

---

The string would be the content of the file. So that it becomes possible to write something like

```
programs.nushell.config = ''
  Whatever
'';
```

and if you want to use a file

```
programs.nushell.config = ./whatever.nu;
```